tests/test_precompiled.py

```diff
@@ -763,3 +763,31 @@ def test_redact_precompiled_letter_address_block_only_touches_first_page():
 
     assert len(doc) == 2
     assert new_second_page_text == second_page_text
+
+
+def test_sanitise_file_contents_on_pdf_with_no_resources_on_one_of_the_pages_content_outside_bounds(
+    client, auth_header
+):
+    """
+    This tests to make sure that PyPDF2 doesn't raise a KeyError when sanitising a PDF that is missing /Resources
+    on one of the pages. Resources should be inferrer from one of the parent/previous pages in this case.
+
+    The PDF under test was provided by one of our services when they encountered the error. Ideally the PDF would
+    be valid and return a successful response, but some test that this returns a correct response is at least better
+    than this call raising an error/500."""
+    with open("tests/test_pdfs/no_resources_on_last_page.pdf", "rb") as pdf:
```

Comment:
You can use the the approach we have for other test PDFs in `pdf_consts.py`, which provides a helper function for opening the file and also means that all test PDFs get documented there.